### ISSUE TITLE
external/msgfmt.py license issue

### ISSUE THREAD

**Issue by kathenas** [jendrikseipp/rednotebook#357]
external/msgfmt.py:

Has no license data in the file. Debian are asking for this.

Also can we confirm that it is GPL-3+ that it states in debian/copyright?

**Comment by jendrikseipp** [comment]
I don't know which license applies to msgfmt.py. You suggested GPL-3+, but I'm not sure that's correct. Let's ask the author directly :-)

@loewis: Under which license is the cpython/Tools/i18n/msgfmt.py script released?

**Comment by kathenas** [comment]
I took it from a quick spin through packages that contained the file in debian and 'calibre' was in the results.

In calibre:

msgfmt.py reside in:

/usr/lib/calibre/calibre/translations/

https://packages.debian.org/stretch/all/calibre/filelist

And in the copyright the contents of that folder is GPL-3.

http://metadata.ftp-master.debian.org/changelogs/main/c/calibre/calibre_2.75.1+dfsg-1_copyright

Bad assumption to rely on another package. :-) Let's wait on the upstream author.

**Comment by jendrikseipp** [comment]
After reading https://github.com/python/cpython#copyright-and-license-information and https://github.com/python/cpython/blob/master/LICENSE, I think we can assume that msgfmt.py is licensed under the PSF License version 2.

**Comment by kathenas** [comment]
I would agree. You made modifications in 2017. So here is the new section for review and approval before I start doing PR.

Python 3.5.3 is the version in debian stable, testing and unstable thus it being the version in this text. 2001-2017 is the dating as 3.5.3 was released early 2017.

Proposed below...

```
Files: rednotebook/external/msgfmt.py
Copyright: Copyright 2001-2017 Python Software Foundation
           Copyright 2017 Jendrik Seipp <jendrikseipp@gmail.com>
License: PSF-2
 1. This LICENSE AGREEMENT is between the Python Software Foundation ("PSF"), and
    the Individual or Organization ("Licensee") accessing and otherwise using Python
    3.5.3 software in source or binary form and its associated documentation.
 .
 2. Subject to the terms and conditions of this License Agreement, PSF hereby
    grants Licensee a nonexclusive, royalty-free, world-wide license to reproduce,
    analyze, test, perform and/or display publicly, prepare derivative works,
    distribute, and otherwise use Python 3.5.3 alone or in any derivative
    version, provided, however, that PSF's License Agreement and PSF's notice of
    copyright, i.e., "Copyright © 2001-2017 Python Software Foundation; All Rights
    Reserved" are retained in Python 3.5.3 alone or in any derivative version
    prepared by Licensee.
 .
 3. In the event Licensee prepares a derivative work that is based on or
    incorporates Python 3.5.3 or any part thereof, and wants to make the
    derivative work available to others as provided herein, then Licensee hereby
    agrees to include in any such work a brief summary of the changes made to Python
    3.5.3.
 .
 4. PSF is making Python 3.5.3 available to Licensee on an "AS IS" basis.
    PSF MAKES NO REPRESENTATIONS OR WARRANTIES, EXPRESS OR IMPLIED.  BY WAY OF
    EXAMPLE, BUT NOT LIMITATION, PSF MAKES NO AND DISCLAIMS ANY REPRESENTATION OR
    WARRANTY OF MERCHANTABILITY OR FITNESS FOR ANY PARTICULAR PURPOSE OR THAT THE
    USE OF PYTHON 3.5.3 WILL NOT INFRINGE ANY THIRD PARTY RIGHTS.
 .
 5. PSF SHALL NOT BE LIABLE TO LICENSEE OR ANY OTHER USERS OF PYTHON 3.5.3
    FOR ANY INCIDENTAL, SPECIAL, OR CONSEQUENTIAL DAMAGES OR LOSS AS A RESULT OF
    MODIFYING, DISTRIBUTING, OR OTHERWISE USING PYTHON 3.5.3, OR ANY DERIVATIVE
    THEREOF, EVEN IF ADVISED OF THE POSSIBILITY THEREOF.
 .
 6. This License Agreement will automatically terminate upon a material breach of
    its terms and conditions.
 .
 7. Nothing in this License Agreement shall be deemed to create any relationship
    of agency, partnership, or joint venture between PSF and Licensee.  This License
    Agreement does not grant permission to use PSF trademarks or trade name in a
    trademark sense to endorse or promote products or services of Licensee, or any
    third party.
 .
 8. By copying, installing or otherwise using Python 3.5.3, Licensee agrees
    to be bound by the terms and conditions of this License Agreement.
```

**Comment by jendrikseipp** [comment]
I wonder whether we shouldn't use the version-agnostic license text from https://github.com/python/cpython/blob/master/LICENSE? 

The copyright shouldn't include me as the change you're referring to only updated the file with the newest version from the Python Git repo.

Since msgfmt.py was created in 2000 (https://github.com/python/cpython/commits/master/Tools/i18n/msgfmt.py), should this be the start of the year range?

**Comment by kathenas** [comment]
Fair enough. I will move things around to make sure that large block of text goes to the bottom of the debian/copyright file.

**Comment by jendrikseipp** [comment]
This issue was fixed in #360 and #361.
